.gitignore

```diff
@@ -9,6 +9,9 @@ obj/
 # Visual Studio
 ###############################################################################
 .vs/
+.dotnet/
```

Comment:
Why? We had them under artifacts, explicitly, in order to help prevent pollution of the root directory...

Reply:
Because it's are not under artifacts anymore. Artifacts are meant for artifacts produced by the build, not by tools the build uses.

---
In reply to: [184185423](https://github.com/dotnet/sdk/pull/2178#discussion_r184185423) [](ancestors = 184185423)

Reply:
These are artifacts produced by the build. They may not be artifacts produced directly by our own source code, but they are artifacts that exist solely to support the build and that are not checked into the source tree. Having it under a single folder makes it very convenient to manage and clean up things, as you have a single location to track, rather than 4-5.

Reply:
`git clean -xfd` cleans everything if that's what you want to do.

Cleaning up the dotnet installation is not common. Deleting artifacts without deleting dotnet is more common. Downloading and installing dotnet takes long time and is only needed when the version changes.


---
In reply to: [184195237](https://github.com/dotnet/sdk/pull/2178#discussion_r184195237) [](ancestors = 184195237)